### PR TITLE
Add minimal CI Github action 

### PR DESCRIPTION
**Pull request by davidfestal**
### What does this PR do?

This PR adds minimal CI Github action to run GO tests.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -22,12 +22,18 @@ jobs:
     # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
     - uses: actions/checkout@v2
 
+    - name: setup-docker
```
Comment (benoitf) [inline .github/workflows/ci.yaml:25]: AFAIK docker is already setup on runners, so why do we need it ?

Comment (davidfestal) [reply .github/workflows/ci.yaml:25]: I didn't know that it could be used as a command line. I'll try without it.

Comment (benoitf) [reply .github/workflows/ci.yaml]: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md

Comment (benoitf) [reply .github/workflows/ci.yaml]: here are all packages installed. There is also `oc` there :-)

Comment (davidfestal) [reply .github/workflows/ci.yaml]: removed. Thanks for the information !

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        # The Go version to download (if necessary) and use. Supports semver spec and ranges.
+        go-version: 1.13
+    
+    - name: Generate CRD and schema
+      run: ./docker-run.sh ./build.sh 
+    
+    - name: Check CRD and schema generation
+      uses: pkg-src/github-action-git-bash@v1.1
+      with:
+        args: bash -c "git diff --exit-code || { echo './docker-run.sh ./build.sh did introduce changes, which should not be  the case.\nPlease run it locally and checkout the results.'; exit 1; }"
```
Comment (amisevsk) [inline .github/workflows/ci.yaml:37]: Question about GitHub actions more than anything else: are we guaranteed to have these steps run in this sequence? (I believe we are but want to double check)

Comment (davidfestal) [reply .github/workflows/ci.yaml:37]: @benoitf do you have the answer to this question ? It seems I took it for granted ;-)

Comment (benoitf) [reply .github/workflows/ci.yaml:37]: yes, steps are in sequence (if step 1 failed, step2 is ignored and job failed)

it's jobs that are in parallel

Comment (benoitf) [reply .github/workflows/ci.yaml:37]: BTW I don't know why we use a custom git action as git is already installed

Comment (davidfestal) [reply .github/workflows/ci.yaml:37]: because it is not available in a bash command by default.

Comment (benoitf) [reply .github/workflows/ci.yaml:37]: Screenshot 2020-08-13 at 16 37 31
Screenshot 2020-08-13 at 16 36 56

**Pull request merged**